### ISSUE TITLE
Submission Query with custom fields does not change the field handle to uid

### ISSUE THREAD

**Issue by Anubarak** [verbb/formie#2060]
### Describe the bug

I am trying to fetch submissions by field handle but it won't work the way you would think it works - I tried to search for the correct place in the docs, but I wasn't able to see how to query for custom fields https://verbb.io/craft-plugins/formie/docs/getting-elements/submission-queries

Usually you would think you could filter submissions via

```PHP
$submission = Submission::find()
    ->form('handle')
    ->myCustomField('foo')
    ->one();
```

however this will just result in an SQL command like
```
(JSON_CONTAINS(formie_submissions.content, '{\"myCustomField\":\"foo\"}'))
```
which is obviously wrong since you need to change the handle to the UID.

I got it working by
```php
$form = Formie::$plugin->getForms()->getFormByHandle('handle');
$layout = $form->getFormLayout();
$field = $layout->getFieldByHandle('myCustomField');

$submission = Submission::find()
    ->form('handle')
    ->andWhere([
        'and',
        ['=', $field->getValueSql(), 'foo'],
    ])
    ->one();
```
However I am not sure if that is really your intended way and if so why it's not in the docs.

### Steps to reproduce

1.  try to create a submission query with a custom field as filter as you would query every other element in Craft
2. see it won't work

### Form settings

does not matter

### Craft CMS version

5.4.0

### Plugin version

3.0.6

### Multi-site?

No

**Comment by engram-design** [comment]
Ah, no that's an issue on our end. Fixed for the next release. To get this early, run `composer require verbb/formie:"dev-craft-5 as 3.0.6"`.

**Comment by engram-design** [comment]
Fixed in [3.0.7](https://github.com/verbb/formie/releases/tag/3.0.7)
